mflowgen/common/scripts/fix-shorts.tcl

```diff
@@ -0,0 +1,132 @@
+########################################################################
+# Save route blockage info in name() box() and layer() arrays
+set i 0
+foreach ptr [get_db route_blockages] {
+    select_obj $ptr
+
+    set blockage_names($i)  [get_db $ptr .name]
+    set blockage_boxes($i)  [get_db $ptr .rects]
+    set blockage_layers($i) [get_db $ptr .layer.name]
+
+    # echo $i $ptr $name($i)
+    # printf "%2d %s %s\n" $i $ptr $name($i)
+    incr i
+}
+set nblockages $i
+
+########################################################################
+# Display list of blockages found e.g.
+#    0 RouteBlk -layer M3 -box {15.93 1670.88 77.225 1898.352} -name def
+#    1 RouteBlk -layer AP -box {0.0 0.0 0.3 1900.416} -name left
+#    2 RouteBlk -layer M9 -box {0.0 0.0 0.3 1900.416} -name left
+#   ...
+#   52 RouteBlk -layer M3 -box {15.93 765.504 77.225 992.976} -name def
+#   53 RouteBlk -layer M3 -box {15.93 539.136 77.225 766.608} -name def
+#   54 RouteBlk -layer M3 -box {15.93 312.816 77.225 540.288} -name def
+#   55 RouteBlk -layer M3 -box {15.93 86.448 77.225 313.92} -name def
+echo "@file_info: Found and saved info for $nblockages route blockages"
+for { set i 0 } { $i < $nblockages } { incr i } {
+    printf "#   %2d " $i
+    echo RouteBlk \
+        -layer $blockage_layers($i) \
+        -box $blockage_boxes($i)    \
+        -name $blockage_names($i)   \
+        ;
+}
+
+########################################################################
+# Delete route blockages (will restore them again later)
+echo "@file_info: Deleting route blockages"
+deleteRouteBlk -all
+redraw; sleep 1
+
+########################################################################
+# Fix shorts 1: Fetch a fresh set of violation markers
+# Note Tile_PE has VDD shorts on lower layers that we choose to ignore...
+# FIXME what's up with the VDD shorts!!??
+clearDrc
+verify_drc -layer_range { M4 M9 }
+
+
+
+########################################################################
+# Fix shorts 2: Identify the shorts
+set shorts [dbGet top.markers { .subType eq "Metal Short" }]
+if { $shorts == "0x0" } {
+    echo "@file_info: No shorts found"
+} {
```

Comment:
if { cond } { then-clause } { else-clause }, yes? No "else" or "then" keywords, right?

Reply:
I've never heard of this syntax. Using else definitely works and is preferred so it's clear what is going on.

Reply:
> I've never heard of this syntax. Using else definitely works and is preferred so it's clear what is going on.

OMG will this merge never end.

Requesting changes like this on working code is not very agile, especially considering that each set of changes is another 32 hours of testing before we get another green light.

The more agile approach is to approve the working code and then file an issue on whatever cleanup is needed. This comment would apply equally to most of your comments that I have addressed so far, e.g. working PHY stripes that should be parameterized instead of hardcoded.

As a result this merge is becoming more and more unwieldy; I vote that *as soon as* we have working code again (see below), we merge immediately and start fresh with a new pull to address any lingering concerns...

Meanwhile it looks like our window has closed again; the latest peak/lassen/magma changes have rendered the PE tile too large again, I will be sending out mail about that soon.

Reply:
> I've never heard of this syntax. Using else definitely works and is preferred so it's clear what is going on.

Now that my rant is out of the way I can address your actual concern. Originally I thought you were mistaken about needing an "else", since my code works, and then I did multiple tests in a tcl shell to confirm that the syntax I was using was doing what I thought it was doing.

Having read the tck/tk manual page https://www.tcl.tk/man/tcl/TclCmd/if.htm I now see that both forms are acceptable. I agree that having the "else" is more readable, and I will try to remember to include it here and elsewhere.

It seems odd to use the "else" without using the "then", but I scanned through all our tcl code so far and that does indeed seem to be the style.